extensions/EXT/EGL_EXT_gl_colorspace_display_p3_passthrough.txt

```diff
@@ -0,0 +1,149 @@
+    is EGL_GL_COLORSPACE_LINEAR.":
+
+    [[ If EGL_EXT_gl_colorspace_display_p3_passthrough is supported ]]
+
+    If its value is EGL_GL_COLORSPACE_DISPLAY_P3_PASSTHROUGH_EXT, then a
+    non-linear, sRGB encoded Display-P3 color space is assumed, with a
+    corresponding GL_FRAMEBUFFER_ATTACHMENT_COLOR_ENCODING value of GL_LINEAR.
+    The application is responsible for applying the appropriate transfer
+    function when writing and reading pixels.
+
+    Modify the 4th paragraph on the same page:
+
+    Note that the EGL_GL_COLORSPACE_DISPLAY_P3_PASSTHROUGH_EXT attribute is
+    used only by OpenGL and OpenGL ES contexts supporting sRGB framebuffers.
+    EGL itself does not distinguish multiple colorspace models.
+    Refer to the "sRGB Conversion" sections of the OpenGL 4.4 and
+    OpenGL ES 3.0 specifications for more information.
+
+    Note that the EGL_GL_COLORSPACE_DISPLAY_P3_PASSTHROUGH_EXT attribute
+    indicates that a colorspace of Display P3 will be communicated to the
+    Window system. While EGL itself is color space agnostic, the surface
+    will eventually be presented to a display device with specific color
+    reproduction characteristics. If any color space transformations are
+    necessary before an image can be displayed, the color space of the
+    presented image must be known to the window system.
+
+Errors
+
+    Modify below error in the "Errors" section on page 34:
+
+    "If config does not support the OpenVG colorspace or alpha format at-
+    tributes specified in attrib list (as defined for eglCreatePlatformWindow-
+    Surface), an EGL_BAD_MATCH error is generated."
+
+    To include OpenGL colorspace as well:
+
+    "If config does not support the OpenGL colorspace, the OpenVG colorspace or
+    alpha format attributes specified in attrib list (as defined for eglCreate-
+    PlatformWindowSurface), an EGL_BAD_MATCH error is generated."
+
+Issues
+
+    1. When creating an EGL surface, what happens when the specified colorspace
+       is not compatible with or supported by the EGLConfig?
+
+       RESOLVED: There is currently no way to query the compatibility of a
+       EGLConfig and colorspace pair. So the only option is to define an error
+       case similar to that of OpenVG colorspace, i.e. if config does not
+       support the colorspace specified in attrib list (as defined for egl-
+       CreateWindowSurface, eglCreatePbufferSurface and eglCreatePixmapSurface),
+       an EGL_BAD_MATCH error is generated.
+
+   2. Why the new enum instead of DISPLAY_P3_EXT + EXT_srgb_write_control?
+
+      RESOLVED: Two aspects, first, EXT_srgb_write_control only affects writes
+      and we want both frame buffer writes and reads to be passthrough. Second,
+      we want to rely on "surface state" rather than a "context state", e.g.
+      EXT_srgb_write_control is global where we only want behavior to apply to
+      specific surface.
+
+   3. Should sRGB framebuffer support affect the pixel path?
```

Comment:
This is from https://www.khronos.org/registry/OpenGL/extensions/ARB/ARB_framebuffer_sRGB.txt.  It says sRGB rendering has no effect on framebuffer reads (glReadPixels, etc.).  This conflicts with the first aspect of the answer to issue 2, which says we need the new enum to disable sRGB framebuffer reads.

I think we can omit issue 3, and the first part of the answer to issue 2.

Reply:
I'll simplify issue #2 to only discuss EXT_srgb_write_control - no mention of frame buffer read / writes.

I want to leave #3 because there were questions internally about whether these extensions affect the readPixels path or not.

Reply:
This is an EGL extension.  The answer should go like "sRGB rendering is defined by GL/GLES.  Specifically, glReadPixels and other pixel paths operations are not affected by sRGB rendering.  But glBlitFramebuffer is."

But I will find this strange because this extension does not enable sRGB rendering.  This belongs to the display_p3 extension.

Reply:
Fixed.